### PR TITLE
Compressing repeated, unwieldy template definitions to make code more…

### PR DESCRIPTION
**Pull request by dmehring**
… readable. No functionality changes. Just macro substitutions to make code more readable.

### CODE REVIEW

File: scimath/Mathematics/ClassicalStatistics.h [hunk 1]
```diff
@@ -38,6 +38,9 @@
 #include <vector>
 #include <utility>
 
+// because the template signature has become unwieldy
+#define _TPARMS AccumType, DataIterator, MaskIterator, WeightsIterator
```
Comment (juliantaylor) [inline scimath/Mathematics/ClassicalStatistics.h]: hm underscore followed by uppercase letter is reserved by the c++ standard for any use (17.6.4.3.2)
I a header it should at least be `#undef` at the end or have a `CASA? string in it

**Comment by dmehring** [comment]
OK. My next set of commits will contain functionality changes. Then after that, I'll fix up the macro so you don't have to review both types of changes at once.

**Comment by juliantaylor** [comment]
can c++11 template aliases be used here? (more out of curiosity as casacore is not c++11 only right now)
maybe one could even use them when the whole template class is a macro in c++98 and a template alias in c++11

**Comment by JimNrao** [comment]
Another option that would reduce the template parameter list usage 
considerably would be to be to include the implementation inside the 
class construct:

template <typename A, typename B, typename C, typename D>
class Classical {

     Classical & operator= (const Classical & other) { ...}

};

vs

template <typename A, typename B, typename C, typename D>
ClassicalOperator<A,B,C,D> & ClassicalOperator<A,B,C,D>::operator=(const 
ClassicalOperator<A,B,C,D> & other) {...}


On 02/05/2016 10:44 AM, Julian Taylor wrote:
>
> can c++11 template aliases be used here? (more out of curiosity as 
> casacore is not c++11 only right now)
> maybe one could even use them when the whole template class is a macro 
> in c++98 and a template alias in c++11
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/casacore/casacore/pull/302#issuecomment-180463358>.Web 
> Bug from 
> https://github.com/notifications/beacon/AGPGtIAgi_S3K8qBNK9rh8kk3U7nsGUDks5phNbugaJpZM4HUbsH.gif
>

-- 
Jim Jacobs

phone: 835-7235
office: 301

**Comment by dmehring** [comment]
committed backward compatible api changes to allow caller to specify number of bins to use in qunatile computations and modified LatticeStatistics implementation to decrease the probability that multiple passes of large images will be necessary when computing quantile-related quantities. I'll work on cleaning up the aliases now. While I understand Jim's point, combining the implementation with the method definitions is not a good idea in this case IMO because this is not generally how we do things and because I think combining large chunks of code with API documentation would make things more difficult to maintain.

**Comment by dmehring** [comment]
committed alias stuff. No functionality changes in this commit.

**Comment by juliantaylor** [comment]
thanks

**Pull request merged**